### PR TITLE
A guide / short intro for ITS Logo versions #1496

### PR DESCRIPTION
**Pull request by Alexey-Slyusar**
Lars, Eric, please take a look at the Logo guide. It is very possible that the document overloads a reader with too much additional info and has some unclear parts. But I tried in this document to introduce the reader with great MIT Logo Group intellectual heritage that we have collected here in some depth.

### CODE REVIEW

**Comment by Alexey-Slyusar** [comment]
I've squashed commits twice. But it seems that it is needed to do only once - right before PR.

**Comment by Alexey-Slyusar** [comment]
I want to add at least one more thing to the document. Although I am not going to change the existing parts of the document.

I'll add the additional part in the morning (US West Coast evening).

File: doc/logo.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+Harold Abelson
+
+"Historically, this idea that Logo is mainly turtle graphics is a mistake. 
+Logo’s name comes from the Greek word for word, because Logo was first
+designed as a language in which to manipulate language: words and sentences."
+
+Brian Harvey
+
+Logo was initially created by Wally Feurzeig, Seymour Papert, and Cynthia
+Solomon in 1967 as part of National Science Foundation sponsored research
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "as part of **a** National ... project"

File: doc/logo.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+project conducted at Bolt, Beranek & Newman, Inc., in Cambridge, Massachusetts.
+In 1969 Dr. Seymour Papert started the Logo Group at the MIT Artificial Intelligence
+Lab. Throughout the 1970s the majority of Logo development was conducted at MIT
+in the Artificial Intelligence Lab and the Division for Study and Research in
+Education, using large research computer systems, such as ITS powered PDP-10.
+
+Our goal is to make that early Logo systems available to wider auditory of
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "to **a** wider auditory"

Maybe audience is a better word than auditory here?

File: doc/logo.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+enthusiasts for exploration, experimenting and, of course, hacking.
+
+### BBN Logo
+
+BBN PDP-10 LOGO system was implemented originally by Walter B. Weiner with
+the assistance of Paul Wexelblat and Charles R. Morgan at BBN. 
+We have source file with a copyright from 1970. It is version modified at National
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "**The** BBN PDP-10 system..."  
"We have **the** source file*  
"It is a version modified at..."

File: doc/logo.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+Research Council, Ottawa, Canada by A.G. Smith and R.A.Orchard.
+That version of BBN Logo for PDP-10 TOPS-10 and TENEX was assembled with MACRO-10
+in ITS, and dumped with the DECUUO bootstrap to get ITS executable binary by Lars
+Brinkhoff.
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "to get **an** ITS executable binary."

I think it's better to remove my name and focus on the historical events.

File: doc/logo.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+That version of BBN Logo for PDP-10 TOPS-10 and TENEX was assembled with MACRO-10
+in ITS, and dumped with the DECUUO bootstrap to get ITS executable binary by Lars
+Brinkhoff. 
+
+to run, type:
+`:bbn;logo`
+
+Here is some BBN Logo related documentation:
+
+ Bolt Beranek and Newman, Inc., Report no. 2165 vol. 4 "The Logo Processor: A Gide
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "G**u**uide"

File: doc/logo.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+ https://github.com/PDP-10/bbn-logo
+ 
+### MIT CLOGO
+
+MIT CLOGO is a direct BBN LOGO descendant. It is, in essence, BBN Logo ported to ITS
+and MIDAS and enhanced with turtle graphics and other capabilities at MIT.
+
+We have pretty stable CLOGO version 49 binary as well as good-commented source code for
+the version.
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "We have **a** pretty stable..."  
"*well*-commented source code"

File: doc/logo.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+Unfinished documentation for the CLOGO may be reached at:
+https://github.com/PDP-10/its/blob/alexey/CLOGO/doc/_info_/clogo.manual
+
+### MIT LISP LOGO
+
+LISP LOGO is an implementation of the LOGO language written in MACLISP for the ITS, TEN50
+and TENEX PDP-10 systems, and MULTICS. The system were implemented by Ira Goldstein,
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "The system **was** implemented"

File: doc/logo.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+Henry Lieberman in the early 70's at MIT.
+One of the reason for this implementation was to provide a natural transition to the more
+powerful computational world of LISP as the Logo programmer grows more sophisticated.
+When desired, the one has access to all of the capabilities of MACLISP.
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "When desired, one has access"

File: doc/logo.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+- BWLOGO - LISP LOGO with black and white graphics.
+
+LISP LOGO had available several packages of special functions:
+- TV TURTLE - LISP LOGO package provides Knight TV system Logo Graphics.
+- DISPLAY TURTLE - LISP LOGO package provides 340 / GT40 display Logo Graphics.
+- GERMLAND - some kind of display turtle for character displays without grahics capabilities.
+- MUSIC BOX - the package provides Logo music capabilities.
+
+The LLOGO variant of LISP LOGO was recently brought back to life during several amazing
+hacking session performed by Eric Swenson and Lars Brinkhoff. As a result, we
```
Comment (larsbrinkhoff) [inline doc/logo.md]: No need to praise Swenson and Brinkhoff here.

File: doc/logo.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+One more powerful feature of LISP LOGO is using the MACLISP compiler directly on LOGO
+source programs and obtain a substantial gain in efficiency, once the programs are
+thoroughly debugged.
+
+Documentation:
+
+[MIT A.I. Memo 307A "LLOGO: An Implementation of LOGO in LISP"](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwjNkYz73JngAhXGkywKHRfMDuQQFjAAegQICRAC&url=https%3A%2F%2Fdspace.mit.edu%2Fbitstream%2Fhandle%2F1721.1%2F6221%2FAIM-307a.pdf&usg=AOvVaw1iwaMXNU6NgcYAFToTxKlY)
+
+[MIT A.I. Memo 361 "The TV Turtle. A Logo Graphics System for Raster Displays"](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=2&cad=rja&uact=8&ved=2ahUKEwj05u7d35ngAhVNl4sKHb2vAIQQFjABegQICRAC&url=https%3A%2F%2Fdspace.mit.edu%2Fbitstream%2Fhandle%2F1721.1%2F5773%2FAIM-361.pdf%3Fsequence%3D2&usg=AOvVaw3reoCDZkG-ZAyhdEe2B4G_)
+
```
Comment (larsbrinkhoff) [inline doc/logo.md]: These two links go through Google.  Please make the direct link to dspace.mit.edu.

File: doc/logo.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+MIT 11LOGO one. It is in essence just logo language evaluator with limited turtle display
+340 grapics capabilities only. A brief description of the built-in procedures (primitives)
+in 11LOGO version that we have is available at `its/doc/_info_/11logo.order document`.
+
+### MIT PLOGO
+
+In accordance with Leigh L Klotz Jr. talk at comp.lang.logo channel 
+"around 1977-1978, Gary Drescher and someone else whose name escapes me at the moment
+wrote a version of Logo in Pascal, as part of a project with Texas Instruments, for
+the TI 99/4 Home Computer, because Pascal was thefonly high-level language supported.
```
Comment (larsbrinkhoff) [inline doc/logo.md]: "Pascal was **the only**"

File: doc/logo.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+
+### MIT APLOGO
+
+In accordance with Leigh L Klotz Jr., Hal Abelson directed the Logo for the Apple II
+project at MIT.
+
+MIT APLOGO was developed by Stephen Hain, Patrick G. Sobalvarro and Leigh L Klotz Jr.
+It was developed and cross-compilled for the Apple-II-Plus Personal Microcomputer on
+PDP-10 at the MIT LOGO Group. It is direct predecessor for Terrapin Logo.
+We have source code for assembling Preliminary Version from 8/06/80 at `its/src/aplogo`
```
Comment (larsbrinkhoff) [inline doc/logo.md]: APLOGO has been updated to an improved 1981 version.

**Comment by larsbrinkhoff** [comment]
Thank you.  There's a wealth of information here, which I think is good.

I made some minor comments.

**Comment by larsbrinkhoff** [comment]
Hello @leighklotz,

Sending this message just to keep you in the loop regarding Logo restoration here.  @Alexey-Slyusar is adding a document to summarize the implementations available.

**Comment by Alexey-Slyusar** [comment]
Lars, I've added a section in the latest commit by `git commit --amend -C HEAD` and did `git push -f`.
Please take a look at this new section at the end of the document too.
And thank you very much for the comments. I am working on it.

I've added  @leighklotz as a reviewer for the document. It will be great if @leighklotz finding some time to take a look at the document. Thanks in advance!

And yes, thank you very much for fearless-git project. It is very helpful. But I still not sure that I do right things with git actually :-)

**Comment by Alexey-Slyusar** [comment]
Oh, it seems that I have not had added @leighklotz as a reviewer actually. I can add only PDP-10 / ITS group members. @larsbrinkhoff I think it will be great to make invite for @leighklotz!

File: doc/logo.md (hunk 1 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 2 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: change for audience.

File: doc/logo.md (hunk 3 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 4 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 5 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 6 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 7 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 8 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

**Comment by larsbrinkhoff** [comment]
Alexey, I believe Mr Klotz will get a message from GitHub when his user name is mentioned with an @.

File: doc/logo.md (hunk 9 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: I do not quite agree, but I obey.

File: doc/logo.md (hunk 10 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 11 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

File: doc/logo.md (hunk 12 above)
Comment (Alexey-Slyusar) [reply doc/logo.md]: done.

**Comment by Alexey-Slyusar** [comment]
I see, Lars, thank you!

**Comment by Alexey-Slyusar** [comment]
I learned that Eric didn't have a chance to take a look at the document. I
believe that it is right idea to wait with the merging until Eric have some
free time to review. We are not in hurry and the quality of the document
benefit a lot from Eric's review.

вс, 17 февр. 2019 г. в 13:40, Lars Brinkhoff <notifications@github.com>:

**Comment by eswenson1** [comment]
@Alexey-Slyusar No, it is not necessary to wait. Feel free to merge. One reviewer is good enough. If I find something I don't like about it, I'll enter a ticket later. Lars approved your PR, so it is good to go.

**Comment by larsbrinkhoff** [comment]
It's not approved yet.

Can you please squash everything together into one single commit and push that?

**Comment by larsbrinkhoff** [comment]
Sorry, I see you did that already.  Can you just update the commit message?

**Comment by Alexey-Slyusar** [comment]
Participating a morning meeting at work. So my office duties swallowed me
up entierly for a while.

пн, 18 февр. 2019 г. в 10:33, Lars Brinkhoff <notifications@github.com>:

**Comment by Alexey-Slyusar** [comment]
Lars, I've updated the commit message.

**Comment by larsbrinkhoff** [comment]
I think the title of this pull request is almost a good commit message.  I updated it and approved.  Alexey, you click the merge button if you think it's ready.

**Comment by Alexey-Slyusar** [comment]
Lars, thank you!!!

**Pull request merged**